tests/test_restserver.py

```diff
@@ -160,10 +242,127 @@ def test_get_block_details(self, mock1, mock2):
         blockhash = get_block_details_blockhash
 
         url = f"/api/block/{blockhash}"
+        client = TestClient(app)
+        response = client.get(url)
+        assert response.json() == block_hash_result
+
+    @mock.patch("opensight_restserver.call_method_electrum")
+    @mock.patch("opensight_restserver.call_method_node")
+    def test_get_block_details_without_value(self, mock1, mock2):
+        mock1.side_effect = [mock_call_node(key="get_block_details")]
+        mock2.side_effect = [block_hash_electrum_result_without_value]
+        blockhash = get_block_details_blockhash
+
+        url = f"/api/block/{blockhash}"
+        client = TestClient(app)
+        response = client.get(url)
+        assert response.json() == block_hash_result_no_reward
+
+    @mock.patch("opensight_restserver.call_method_electrum")
+    @mock.patch("opensight_restserver.call_method_node")
+    def test_get_block_details_value_satoshi(self, mock1, mock2):
+        mock1.side_effect = [mock_call_node(key="get_block_details")]
+        mock2.side_effect = [mock_electrum_connect(key="get_block_details_value_satoshi")]
+        blockhash = get_block_details_blockhash
+
+        url = f"/api/block/{blockhash}"
+        client = TestClient(app)
+        response = client.get(url)
+        assert response.json() == block_hash_result
+
+    @mock.patch("opensight_restserver.call_method_node")
+    def test_get_block_details_no_block_found(self, mock1):
+        mock1.side_effect = [""]
+        blockhash = get_block_details_blockhash
+
+        url = f"/api/block/{blockhash}"
+        client = TestClient(app)
+        response = client.get(url)
+
+        assert response.json() == {'message': 'block id not found'}
+        assert response.status_code == 404
+
+    def test_log_if(self):
+        log("beat_test_log_lol_>////<")
+
+    def test_log_else(self):
+        logger = logging.getLogger(__name__)
+        log("beat_test_log_lol_>////<", logger)
+
+    @mock.patch("opensight_restserver.call_method_node")
+    def test_get_tx_details_404(self, mock1):
+        mock1.side_effect = [
+            mock_call_node(key = "empty")
+        ]
+        result, status = get_tx_details("thank")
+        assert result == "Not found"
+        assert status == 404
+
+    @mock.patch("opensight_restserver.call_method_node")
+    def test_format_tx_vin(self, mock1):
+        mock1.side_effect = [
+            mock_call_node(key="transaction_details_2")
+        ]
+        vin = {
+            "coinbase": " ",
+            "txid": " ",
+            "valueSat": " ",
+            "vout": 0,
+            "value": " ",
+            "cashAddress": " ",
+            "scriptPubKey": " ",
+            "addresses": " ",
+            "doubleSpentTxID": " "
+        }
+        n = " "
+        format_tx_vin(vin, n)
+
```

Comment:
assert something here, this test isnt testing against anything, only executing the lines of code.